src/split_at.rs

```diff
@@ -233,59 +213,611 @@ unsafe impl<T> SplitAt for [T] {
     }
 }
 
-/// Splits `T` in two.
+/// A `T` that has been split into two possibly-overlapping parts.
 ///
-/// # Safety
-///
-/// The caller promises that:
-/// 0. `l_len.get()` is not greater than the length of `ptr`'s trailing slice.
-/// 1. if `I::Aliasing` is [`Exclusive`] or `T` permits interior mutation, then
-///    `l_len.padding_needed_for() == 0`.
-#[inline(always)]
-unsafe fn ptr_split_at_unchecked<'a, T, I, R>(
-    ptr: Ptr<'a, T, I>,
-    l_len: MetadataOf<T>,
-) -> (Ptr<'a, T, I>, Ptr<'a, [T::Elem], I>)
+/// For some dynamically sized types, the padding that appears after the
+/// trailing slice field [is a dynamic function of the trailing slice
+/// length](KnownLayout#slice-dst-layout). If `T` is split at a length that
+/// requires trailing padding, the trailing padding of the left part of the
+/// split `T` will overlap the right part. If `T` is a mutable reference or
+/// permits interior mutation, you must ensure that the left and right parts do
+/// not overlap. You can do this at zero-cost using using
+/// [`Self::via_immutable`], [`Self::via_into_bytes`], or
+/// [`Self::via_unaligned`], or with a dynamic check by using
+/// [`Self::via_runtime_check`].
+#[derive(Debug)]
+pub struct Split<T> {
+    /// A pointer to the source slice DST.
+    ///
+    /// # Safety
+    ///
+    /// `source` is a pointer to a slice DST; see also `Self::l_len`.
```

Comment:
This is redundant since we know at each impl site that `T = &U` or `T = &mut U`, and that `U: SplitAt`, which implies `U: KnownLayout<PointerMetadata = usize>`. Instead, we should remove this and have `l_len` promse that "If `source` is a pointer to a slice DST, then...".

Reply:
https://github.com/google/zerocopy/compare/b21d06c408367bc92de75f99017fa67dad9e8aa4..187c1794711c6487fe424506c9132cf9b92f8576#diff-394d6458153031257865a7aaddcda3ac55a4ae61c642dee241136b08a66fcf90R236